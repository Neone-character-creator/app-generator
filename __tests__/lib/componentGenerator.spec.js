const componentGenerator = require("../../lib/componentGenerator");
const sinon = require("sinon");
const renderer = require("react-test-renderer");

describe("component generator module", () =>{
    let componentWriter;
    beforeEach(()=>{
        componentWriter = sinon.stub()
    });

    it("writes an app component", async () => {
        const hierarchy = {
            appName: "test",
            views: [
                "one"
            ],
            components: {
                one: {
                    type: "view",
                    children: ["foo"]
                },
                foo: {
                    type: "textfield"
                }
            }
        };
        const components = await componentGenerator(hierarchy);
        expect(components.app.path).toBe('components/App.js');
        expect(renderer.create(components.app.content).toJSON()).toMatchSnapshot();

        expect(components.views.one.path).toBe('components/OneView.js');
        expect(renderer.create(components.views.one.content).toJSON()).toMatchSnapshot();
    });
});