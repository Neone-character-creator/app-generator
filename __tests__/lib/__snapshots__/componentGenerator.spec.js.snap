// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component generator module writes an app component 1`] = `
"import React from \\"react\\";
import { connect } from \\"react-redux\\";
import OneView from \\"./components/OneView\\"

const Component = () => (
    <div id=\\"test-app\\">
        <OneView/>
    </div>
);

export default Component;"
`;

exports[`component generator module writes an app component 2`] = `
"import React from \\"react\\";
import { connect } from \\"react-redux\\";
import TextField from \\"@material-ui/core/TextField\\";

const Component = () => (
    <div id=\\"one-view\\">
<TextField label=\\"foo\\" id=\\"foo\\" value=\\"\\" />
    </div>
);

export default Component;"
`;
